manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/services/SinkConnectorService.java

```diff
@@ -0,0 +1,78 @@
+package com.redhat.service.smartevents.manager.v2.services;
+
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import com.redhat.service.smartevents.infra.core.models.connectors.ConnectorType;
+import com.redhat.service.smartevents.infra.v2.api.V2APIConstants;
+import com.redhat.service.smartevents.infra.v2.api.models.ComponentType;
+import com.redhat.service.smartevents.infra.v2.api.models.ConditionStatus;
+import com.redhat.service.smartevents.infra.v2.api.models.DefaultConditions;
+import com.redhat.service.smartevents.manager.v2.ams.QuotaConfigurationProvider;
+import com.redhat.service.smartevents.manager.v2.api.user.models.responses.ConnectorResponse;
+import com.redhat.service.smartevents.manager.v2.api.user.models.responses.SinkConnectorResponse;
+import com.redhat.service.smartevents.manager.v2.persistence.dao.ConnectorDAO;
+import com.redhat.service.smartevents.manager.v2.persistence.dao.SinkConnectorDAO;
+import com.redhat.service.smartevents.manager.v2.persistence.models.Condition;
+import com.redhat.service.smartevents.manager.v2.persistence.models.Connector;
+
+@ApplicationScoped
+public class SinkConnectorService extends AbstractConnectorService {
+
+    private static final String URI_DSL_PREFIX = "knative:endpoint/ob-";
+
+    @Inject
+    SinkConnectorDAO sinkConnectorDAO;
+
+    @Inject
+    QuotaConfigurationProvider quotaConfigurationProvider;
+
+    @Override
+    protected ConnectorType getConnectorType() {
+        return ConnectorType.SOURCE;
+    }
+
+    @Override
+    protected ConnectorDAO getDAO() {
+        return sinkConnectorDAO;
+    }
+
+    @Override
+    protected long getOrganisationConnectorsQuota(String organisationId) {
+        return quotaConfigurationProvider.getOrganisationQuotas(organisationId).getSinkConnectorsQuota();
+    }
+
+    @Override
+    protected List<Condition> createAcceptedConditions() {
+        List<Condition> conditions = new ArrayList<>();
+        conditions.add(new Condition(DefaultConditions.CP_KAFKA_TOPIC_CONNECTOR_READY_NAME, ConditionStatus.UNKNOWN, null, null, null, ComponentType.MANAGER, ZonedDateTime.now(ZoneOffset.UTC)));
```

Comment:
I know it's not your doing; but IMO using a constructor with `null`'s suggests we should have a different constructor.

Reply:
Yep, I agree. I'll create a ticket for that :+1:

Reply:
https://issues.redhat.com/browse/MGDOBR-1406